src/main/java/org/sharedid/endpoint/handler/CheckGdprConsentHandler.java

```diff
@@ -0,0 +1,176 @@
+        this.consentGivenByConsentStringMeter = metricRegistry.meter(METRIC_CONSENT_GIVEN_BY_CONSENT_STRING);
+        this.consentNotGivenByGdprParameterMeter = metricRegistry.meter(METRIC_CONSENT_NOT_GIVEN_BY_GDPR_PARAMETER);
+        this.requestInEeaCountryMeter = metricRegistry.meter(METRIC_REQUEST_IN_EEA_COUNTRY);
+        this.requestInNonEeaCountryMeter = metricRegistry.meter(METRIC_REQUEST_IN_NON_EEA_COUNTRY);
+        this.consentStringFromRequestParameterMeter = metricRegistry.meter(METRIC_CONSENT_STRING_FROM_REQUEST_PARAMETER);
+        this.consentStringNotFoundMeter = metricRegistry.meter(METRIC_CONSENT_STRING_NOT_FOUND);
+    }
+
+    @Override
+    public void handle(RoutingContext routingContext) {
+        DataContext dataContext = DataContext.from(routingContext);
+
+        HttpServerResponse response = routingContext.response();
+
+        hasConsent(dataContext, routingContext.request()).setHandler(consentResult -> {
+            if (consentResult.failed()) {
+                logger.debug("Failed to check for consent {}", consentResult.cause().getMessage());
+                consentCheckFailedMeter.mark();
+                response.setStatusCode(204);
+                response.end();
+                return;
+            }
+
+            if (!consentResult.result()) {
+                logger.debug("Consent not given");
+                consentNotGivenMeter.mark();
+
+                //does not have consent
+                //do not redirect
+                //do not write cookies
+
+                response.setStatusCode(204);
+                response.end();
+                return;
+            }
+
+            logger.debug("Consent given");
+            consentGivenMeter.mark();
+
+            routingContext.next();
+        });
+    }
+
+    private Future<Boolean> hasConsent(DataContext dataContext, HttpServerRequest request) {
+        String gdprConsentParam = dataContext.getGdprConsentParam();
+        GdprConsentString gdprConsentString = dataContext.getGdprConsentString();
+
+        if (StringUtils.isBlank(gdprConsentParam)) {
+            if (hasAuditCookie(dataContext)) {
+                //assume that consent was given previously
+                return Future.succeededFuture(true);
+            }
+
+            consentStringNotFoundMeter.mark();
+        } else {
+            consentStringFromRequestParameterMeter.mark();
+        }
+
+        boolean isConsentGiven = true;
+
+        if (gdprConsentString != null) {
+            isConsentGiven = gdprConsentString.isConsentGiven(SHARED_ID_VENDOR_ID);
+        } else {
+            if (StringUtils.isNotBlank(gdprConsentParam)) {
+                //consent string is malformed, consent is not given
+                logger.debug("Gdpr consent string malformed");
+                isConsentGiven = false;
+            } else {
+                logger.debug("Gdpr consent string missing");
+            }
+        }
+
+        if (isConsentGiven) {
```

Comment:
If I understand the code correct, when `dataContext.getGdprConsentParam()` is blank, `dataContext.getGdprConsentString()` will be blank as well. So `isConsentGiven` will set to true, and it will return `Future.succeededFuture(true)` inside this block. But I think it supposes do further check `dataContext.isGdprParam()` and Geo check.

Reply:
Yes, that was an oversight. I updated to check if request comes from a GDPR country if the string is missing.

Reply:
cool.